cpp/src/strings/utilities.cu

```diff
@@ -158,8 +158,13 @@ int64_t get_offset64_threshold()
 
 bool is_large_strings_enabled()
 {
+  // default depends on compile-time switch but can be overridden by the environment variable
   auto const env = std::getenv("LIBCUDF_LARGE_STRINGS_ENABLED");
+#ifdef CUDF_LARGE_STRINGS_DISABLED
   return env != nullptr && std::string(env) == "1";
+#else
+  return env == nullptr || std::string(env) == "1";
```

Comment:
It seems weird to me that the mere existence of the `LIBCUDF_LARGE_STRINGS_ENABLED` isn't how we signal we want support. 

That would allow us to remove the value check, and supports use cases where somebody has done `export LIBCUDF_LARGE_STRINGS_ENABLED`

Reply:
I don't think that is how environment variables work.
```
:~$ export LIBCUDF_LARGE_STRINGS_ENABLED
:~$ echo $LIBCUDF_LARGE_STRINGS_ENABLED

:~$
```
They must be set to a value or they do not exist.

Reply:
> They must be set to a value or they do not exist. 

They can exist and not have a value

```
$ export LIBCUDF_LARGE_STRINGS_ENABLED
$ env | grep LIBCUDF_LARGE_STRINGS_ENABLED
LIBCUDF_LARGE_STRINGS_ENABLED=

$ echo ${LIBCUDF_LARGE_STRINGS_ENABLED+1}
1

$ env | grep FOO
$ 
$ echo ${FOO+1}
$ 
```

We use parameter expansion to verify this ( https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html#Shell-Parameter-Expansion )

Reply:
Ok, this seems very odd to me. So which oddness is less common?
I like `LIBCUDF_LARGE_STRINGS_ENABLED=1` means enable and `LIBCUDF_LARGE_STRINGS_ENABLED=0` is disable and no variable means whatever it was compiled as. Easy to document and predict.

Reply:
Env variables do not seem to work as this on Windows.
```
C:\>set LIBCUDF_LARGE_STRINGS_ENABLED
Environment variable LIBCUDF_LARGE_STRINGS_ENABLED not defined
```

Reply:
Looking at https://docs.nvidia.com/cuda/cuda-c-programming-guide/#cuda-environment-variables I see all the variables expecting a value, so the current approach seems good

Reply:
Also, it does not work with the `/bin/sh` (Bourne shell I believe):
```
$ export LIBCUDF_LARGE_STRINGS_ENABLED
$ echo $LIBCUDF_LARGE_STRINGS_ENABLED

$ env | grep LIBCUDF_LARGE_STRINGS_ENABLED
$      
$ echo ${LIBCUDF_LARGE_STRINGS_ENABLED+1}

$ echo ${FOO+1}

$ echo $((LIBCUDF_LARGE_STRINGS_ENABLED+1))
1
$ echo $((FOO+1))
1
```